tests/providers/test_adapter.py

```diff
@@ -853,6 +853,39 @@ def test_kraus_braket_bit_ordering(self):
         res = LocalSimulator("braket_dm").run(qqc, shots=1000).result().measurement_counts
         assert res["01"] == 1000
 
+    def test_noncontiguous_qubit_device_compatibility(self):
+        """Tests that devices with non-contiguous qubit numbering work correctly."""
```

Comment:
It's unclear from the test that non-contiguous bits are supported or generated for the device.